components/raftstore-v2/src/operation/command/admin/split.rs

```diff
@@ -64,20 +74,387 @@ pub struct SplitResult {
+        self.post_split();
+
+        // It's not correct anymore, so set it to false to schedule a split
+        // check task.
+        self.set_may_skip_split_check(false);
+
+        let is_leader = self.is_leader();
+        if is_leader {
+            self.set_approximate_size(estimated_size);
+            self.set_approximate_keys(estimated_keys);
+            // todo: this may should be doing in handle_peer_split_response.
+            self.heartbeat_pd(store_ctx);
+
+            info!(
+                self.logger,
+                "notify pd with split";
+                "region_id" => self.region_id(),
+                "peer_id" => self.peer_id(),
+                "split_count" => regions.len(),
+            );
+
+            // todo: report to PD
+        }
+
+        assert!(self.split_progress_mut().is_empty());
+        let last_key = enc_end_key(regions.last().unwrap());
+        if meta.region_ranges.remove(&last_key).is_none() {
+            panic!("{:?} original region should exist", self.logger.list());
+        }
+        let last_region_id = regions.last().unwrap().get_id();
+        for (new_region, locks) in regions.into_iter().zip(region_locks) {
+            let new_region_id = new_region.get_id();
+
+            if new_region_id == region_id {
+                let not_exist = meta
+                    .region_ranges
+                    .insert(enc_end_key(&new_region), new_region_id)
+                    .is_none();
+                assert!(not_exist, "[region {}] should not exist", new_region_id);
+                continue;
+            }
+
+            let mut raft_message = self.prepare_raft_message();
+            raft_message.mut_from_peer().set_id(raft::INVALID_ID);
+            raft_message.set_region_id(new_region_id);
+            raft_message.set_to_peer(
+                new_region
+                    .get_peers()
+                    .iter()
+                    .find(|p| p.store_id == store_ctx.store_id)
+                    .unwrap()
+                    .clone(),
+            );
+
+            self.split_progress_mut().insert(new_region_id, false);
+            let init_info = SplitRegionInitInfo {
+                region: new_region,
+                parent_region_id: derived_id,
+                parent_epoch: derived_epoch.clone(),
+                parent_is_leader: self.is_leader(),
+                parent_stat: self.peer_stat().clone(),
+                approximate_keys: estimated_keys,
+                approximate_size: estimated_size,
+                locks,
+                last_split_region: last_region_id == new_region_id,
+            };
+
+            store_ctx
+                .router
+                .send_control(StoreMsg::CreatePeer(Box::new(CreatePeer {
+                    raft_message,
+                    split_region_info: init_info,
+                })));
+        }
+        drop(meta);
+
+        if is_leader {
+            self.on_split_region_check_tick();
+        }
```

Comment:
missing the failpoint " fail_point!("after_split", self.ctx.store_id() == 3, |_| {});"